ucsb-cs156-s22/s22-6pm-happycows
# s22-6pm-3 Changed Columns in LeaderboardTable.js + Adopted those Changes to Tests and Storybook + Fixed Bug with Profits

# Overview
Yvonne removed 'cowHealth' and 'fullName'  from `Leaderboard.js` and added a 'userId' column to the table. 
Yvonne created a new file `leaderboardFixtures.js` that has lists of userCommons with 'userId' and 'commonsId' based on the userCommons entity for corrected testing and storybook.
Yvonne edited `LeaderboardTable.stories.js` and `LeaderboardTable.test.js` to reflect the changes to the columns.

Ayush added a function to profitscontroller.java to facilitate the backend request using commonsId.
Ayush fixed an error in the  PlayPage.js where it was incorrectly making a backend request using the wrong variable(using commonsId instead of userCommonsId).

# User Stories
- As a user, I can view the list of users and just their stats (ie. userId, numCows, wealth) in any commons that I am currently in.
- As an admin, I can view the list of users and their stats+usercommonsid in any commons that I am currently in.

# Issues Addressed
Closes #11

# Code details / modifications:
- Edited `LeaderboardTable.js` where I created a table that includes columns: 'User Id', 'Cows Owned', and 'Wealth', and '(Admin) userCommons Id' if you are an admin.
- Altered `LeaderboardTable.stories.js` where on Storybook, I created 7 instances of the Leaderboard table. One with nothing in it, a second with 1 user, a third with 3 users, a fourth with 5 users, a fifth with 10 users, a sixth with 3 users viewing as admin, and a seventh with 5 users viewing as admin. I used the `leaderboardFixtures.js` instead of `userCommonsFixtures.js`
- Changed `LeaderboardTable.test.js` where it tests the expected outputs for a user and admin when viewing the Leaderboard table in various states (empty or filled).
- Fixed an error in the  PlayPage.js where it was incorrectly making a backend request using the wrong variable(using commonsId instead of userCommonsId)
- Added a function to profitscontroller.java to facilitate the backend request using commonsId

# Test Plan (for interactive testing)

* Load the page onto localhost:8080
* Run npm run storybook and view how a user/admin should view the table.
* Join a user commons and see that their is no toast error when visiting a commons on heroku

Code Coverage:
![image](https://user-images.githubusercontent.com/56051313/171557382-9acf3202-48f6-43ba-a151-021f61a0968a.png)

Mutation Testing:
![image](https://user-images.githubusercontent.com/56051313/171557406-a288d91a-cd1b-4d9f-8813-923a2f9586c7.png)

# Notes!
- An additional feature that could be added to `LeaderboardTable.js` would be to call the backend and use the 'userId' to grab and display their full name. However the issue would be how to reflect 'calling the backend' when using fixtures to test and create stories.

# Further Details 
**Storybook Demonstration w/ ten userCommons as an Admin**
![image](https://user-images.githubusercontent.com/56051313/171557366-86d1bbd5-76c2-4e9c-ac9e-e5921c8a89af.png)
